src/builtins.ts

```diff
@@ -2718,8 +2718,8 @@ export function compileAllocate(
   var allocatePrototype = program.elementsLookup.get(allocateInternalName);
   if (!allocatePrototype) {
     program.error(
-      DiagnosticCode.Cannot_find_name_0,
-      reportNode.range, allocateInternalName
+      DiagnosticCode.Cannot_allocate_memory_unless_an_allocator_is_defined_Please_define_an_allocator_e_g_by_importing_allocator_arena_or_allocator_tlsf,
```

Comment:
What do you think of retaining the original error message for compatibility with tooling, but additionally emitting an info message with more information?

Reply:
Makes sense. I guess it feels a bit weird to say that `allocate_memory` isn't defined when it's never explicitly referenced, and maybe this should be seen as fundamentally different from a "cannot find name" error (especially since it's already a special case in the compiler). But it looks like TypeScript gives a regular `'React' is not defined` error when using JSX without having imported React, which is pretty much the same situation, so interpreting it as the more general error makes sense as long as it's not confusing.